test/functional/rpc_rawtransaction.py

```diff
@@ -176,6 +176,28 @@ def run_test(self):
         self.nodes[0].reconsiderblock(block1)
         assert_equal(self.nodes[0].getbestblockhash(), block2)
 
+        # Create new transaction and don't broadcast.
+        to_address = self.nodes[0].getnewaddress()
+        prevout_index = [i for i, out in enumerate(gottx['vout']) if out['value'] == 1][0]
+        inputs = [{"txid": tx, "vout": prevout_index}]
+        outputs = {to_address: 0.9999}
+        spending_rawtx = self.nodes[1].createrawtransaction(inputs, outputs)
+        spending_tx = self.nodes[1].signrawtransactionwithwallet(spending_rawtx)['hex']
+        spending_txid = bytes_to_hex_str(hash256(hex_str_to_bytes(spending_tx)))
+
+        # Searching unknown transaction with out txindex.
+        assert_raises_rpc_error(-33, "No such mempool transaction. Use -txindex to enable blockchain transaction queries.", self.nodes[1].getrawtransaction, spending_txid)
+
+        # Searching unknown transaction with txindex.
+        assert_raises_rpc_error(-5, "No such mempool or blockchain transaction.", self.nodes[2].getrawtransaction, spending_txid)
+
+        # Broadcast transaction to mempool.
+        self.nodes[0].sendrawtransaction(spending_tx)
```

Comment:
IMO this could be removed (lines 193 to 200), does't test error messages.

Reply:
The purpose of the assertion on line 199 is to test that the error received on 192 goes away after transaction is in the mempool.

Reply:
Right, a comment there would be nice then, like:
```py
# No error should be raised now that the transaction is known.
```